### PR TITLE
scrypt: initial integration

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR initialises OSS-Fuzz integration for the scrypt project in Rust. New fuzzers have been created, and a PR (https://github.com/RustCrypto/password-hashes/pull/534) has been submitted upstream to merge the fuzzers.

### CODE REVIEW

**Comment by arthurscchan** [comment]
@newpavlov. Thanks for merging the fuzzers upstream, could we use the email from your GH page as contact, such that you will receive notifications if issues are found?

**Comment by newpavlov** [comment]
@arthurscchan
Yes, you can. It's also worth to add @tarcieri as well.

While we at it, it would be nice to fix in a separate PR the old job referenced [here](https://github.com/RustCrypto/password-hashes/pull/534#issuecomment-2427731785). I've unsubscribed from it because of continuous notifications, so after fixing it, my email probably should be re-added.

File: projects/scrypt/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Build the fuzzers and project source code
+cargo fuzz build
```
Comment (newpavlov) [inline projects/scrypt/build.sh:19]: Maybe it's worth to use `--release` here? We can enable debug checks for release profile in fuzz/Cargo.toml.

**Review by newpavlov — APPROVED** [review]

File: projects/scrypt/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Copy built fuzzer binaries to $OUT
+find $SRC/password-hashes/fuzz/target/x86_64-unknown-linux-gnu/release -maxdepth 1 \
+    -type f -perm -u=x -exec cp {} $OUT \;
```
Comment (newpavlov) [inline projects/scrypt/build.sh:23]: What will happen, if in the future we add fuzz targets for other crates? If it runs all generated binaries, then I think it's better to rename the folder from `projects/scrypt` to `projects/password-hashes`.

Comment (DavidKorczynski) [reply projects/scrypt/build.sh:23]: I think we'd prefer to have this an explicit connection to the scrypt crate and keep the scrypt project name. Would you prefer to make the harness selection more constrained in this case?

Comment (arthurscchan) [reply projects/scrypt/build.sh:23]: I actually make this dynamic to allow adding more fuzzers for the the scrypt crate without the need to modify the OSS-Fuzz integration later. But I do agree that using specific cp command is also OK. I am OK with either way.

Comment (newpavlov) [reply projects/scrypt/build.sh:23]: In that case it's worth to copy only binaries which satisfy the `scrypt*` regexp.

Comment (arthurscchan) [reply projects/scrypt/build.sh:23]: Thanks for the suggestion. I have fixed that.

**Review by newpavlov — APPROVED** [review]

**Pull request merged**